IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -870,6 +1014,89 @@ - (BOOL)clearWithContext:(id<MSIDRequestContext>)context
+    
+    if (status != errSecSuccess)
+    {
+        [self createError:@"Failed to create SecAccessRef for current application. Please make sure the app you're running is properly signed."
+                   domain:MSIDKeychainErrorDomain errorCode:status error:error context:nil];
+         MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to create SecAccessRef for current application. Please make sure the app you're running is properly signed (status: %d).", (int)status);
+        return nil;
+    }
+    
+    if (![self accessSetACLTrustedApplications:access
+                           aclAuthorizationTag:kSecACLAuthorizationChangeACL
+                           trustedApplications:trustedApplications
+                                       context:nil
+                                         error:error])
+    {
+        CFReleaseNull(access);
+        return nil;
+    }
+    
+    return CFBridgingRelease(access);
+}
+
+- (BOOL)accessSetACLTrustedApplications:(SecAccessRef)access
+                     aclAuthorizationTag:(CFStringRef)aclAuthorizationTag
+                     trustedApplications:(NSArray<id> *)trustedApplications
+                                 context:(id<MSIDRequestContext>)context
+                                   error:(NSError **)error
+{
+    NSArray *acls = (__bridge_transfer NSArray*)SecAccessCopyMatchingACLList(access, aclAuthorizationTag);
+    OSStatus status;
+    CFStringRef description = nil;
+    CFArrayRef oldtrustedAppList = nil;
+    SecKeychainPromptSelector selector;
+    
+    // TODO: handle case where tag is not found?
+    for (id acl in acls)
+    {
+        status = SecACLCopyContents((__bridge SecACLRef)acl, &oldtrustedAppList, &description, &selector);
+        
+        if (status != errSecSuccess)
+        {
+            [self createError:@"Failed to get contents from ACL. Please make sure the app you're running is properly signed." domain:MSIDKeychainErrorDomain errorCode:status error:error context:context];
+             MSID_LOG_WITH_CTX(MSIDLogLevelError, context, @"Failed to get contents from ACL. Please make sure the app you're running is properly signed (status: %d).", (int)status);
+            return NO;
+        }
+        
+        status = SecACLSetContents((__bridge SecACLRef)acl, (__bridge CFArrayRef)trustedApplications, description, selector);
+        
+        if (status != errSecSuccess)
+        {
+            [self createError:@"Failed to set contents for ACL. Please make sure the app you're running is properly signed." domain:MSIDKeychainErrorDomain errorCode:status error:error context:context];
+            MSID_LOG_WITH_CTX(MSIDLogLevelError, context, @"Failed to set contents for ACL. Please make sure the app you're running is properly signed (status: %d).", (int)status);
+            return NO;
+        }
+    }
+    
+    CFReleaseNull(oldtrustedAppList);
```

Comment:
what if SecACLCopyContents succeeds and sets not-null oldtrustedApplist and description, but SecACLSetContents fails? Should we release those in that case as well?

Reply:
Addressed,